lib/topological_inventory/orchestrator/worker.rb

```diff
@@ -1,97 +1,79 @@
 require "json"
-require "yaml"
-require "rest-client"
+require "manageiq-password"
 require "pg"
-require "miq-password"
+require "rest-client"
+require "yaml"
 
+require "topological_inventory/orchestrator/object_database"
 require "topological_inventory/orchestrator/object_manager"
 
 module TopologicalInventory
   module Orchestrator
     class Worker
-      attr_reader :api_url, :ingress_url
-
-      def initialize(api_url, ingress_url, collector_definitions_file)
-        raise "Access to the cluster using serviceaccounts is required" unless ObjectManager.available?
-        @api_url               = api_url
-        @ingress_url           = ingress_url
-        @object_manager        = ObjectManager.new
-        @collector_definitions = YAML.load_file(collector_definitions_file)
+      def initialize(api_base_url: ENV["API_URL"], collector_definitions_file: ENV["CDF"])
+        @api_base_url = api_base_url
+        @collector_definitions_file = collector_definitions_file || TopologicalInventory::Orchestrator.root.join("config/collector_definitions.yaml")
       end
 
       def run
         loop do
-          Signal.trap("TERM") { break }
-          collector_definitions.each do |source_type, endpoint_info|
-            endpoint_info.each do |endpoint_role, options|
-              resolve_collector_type(source_type, endpoint_role, options)
-            end
-          end
+          make_openshift_match_database
+
           sleep 10
         end
       end
 
-      def resolve_collector_type(source_type, endpoint_role, options)
-      end
+      private
 
-      def create_collectors_for_new_sources
-        sources.each do |source|
-          create_objects_for_source(source) unless have_deployment_for_source?(source)
-        end
-      end
-      
-      def remove_collectors_for_deleted_sources
-        collector_deployments.each do |deployment|
-          unless sources.include?(source_for_deployment(deployment))
-            @object_manager.delete_deployment_config(deployment.metadata.name)
-            @object_manager.delete_secret("#{deployment.metadata.name}-secrets")
+      def make_openshift_match_database
+        database = ObjectDatabase.new
+        expected = collectors_from_database(database)
+        current  = collectors_from_openshift
+
+        puts "Checking..."
+
+        (current - expected).each { |i| remove_openshift_objects_for_source(i) }
+        (expected - current).each { |i| create_openshift_objects_for_source(i, database[i]) }
+
+        puts "Checking... complete."
+      end
+
+
+      ### API STUFF
```

Comment:
Follow up, but instead of these comments should these be separate objects? Maybe we could compose these functions better.